ir/value.cpp

```diff
@@ -110,4 +111,37 @@ expr Input::getTyVar() const {
   return expr::mkVar(tyname.c_str(), 2);
 }
 
+
+static string genVectorName(const std::vector<Value *> &vals) {
+  ostringstream ss;
+  ss << "<";
+  for (unsigned idx = 0; idx < vals.size(); ++idx) {
+    ss << *vals[idx];
+    if (idx != vals.size() -1)
+      ss << ", ";
+  }
+  ss << ">";
+  return ss.str();
+}
+
+VectorValue::VectorValue(Type &type, std::vector<Value *> &&vals)
+  : Value(type, genVectorName(vals)), vals(std::move(vals)) {}
+
+void VectorValue::print(ostream &os) const {
+  UNREACHABLE();
+}
+
+StateValue VectorValue::toSMT(State &s) const {
+  expr val;
+  expr not_poison = true;
+
+  for (unsigned idx = 0; idx < vals.size(); ++idx) {
+    auto [v, vs] = vals[idx]->toSMT(s);
+    val = idx == 0 ? v : val.concat(v);
```

Comment:
float values need to be cast to bit-vectors.
I would suggest we introduce a new function in the VectorType that creates an SMT value from a vector of individual values (to avoid repeating this code to cast to BV, as it will probably needed later).